### ISSUE TITLE
Freeze with “Error running timer ‘imenu-list-update’“

### ISSUE THREAD

**Issue by tastytea** [bmag/imenu-list#72]
I experience frequent freezes with lsp-mode and imenu-list:

```
Error running timer ‘imenu-list-update’: (error "Timeout while waiting for response. Method: textDocument/documentSymbol.")
```

It seems to happen whenever company tries to show completions. The freezes happen beginning with 5347c67fe3c48417ade1404472a7bb99670b4001.

Interestingly, the problem goes away when I run `normal-mode` before I type anything.

When I run `imenu-list-stop-timer` it doesn't freeze anymore.

**Comment by bmag** [comment]
Thanks for the report. I don't know why Company completion would trigger imenu-list's update, but it's possible that imenu-list's idle timer just happens to trigger while you are waiting for Company completion to start. Anyway, can you test #73 and tell me if it fixes the issue? It fixes the only thing the looked suspicious to me in https://github.com/bmag/imenu-list/commit/5347c67fe3c48417ade1404472a7bb99670b4001 - setting `imenu-list-auto-update` via Custom would start the timer even if `imenu-list-minor-mode` is disabled.

**Comment by tastytea** [comment]
I'm not sure if it was really company – it was just an educated guess.

#73 fixes the issue, thanks!

**Comment by bmag** [comment]
Merged #73, new version will be available from Melpa shortly.
